mllib/src/main/scala/org/apache/spark/mllib/feature/PCA.scala

```diff
@@ -44,6 +44,11 @@ class PCA @Since("1.4.0") (@Since("1.4.0") val k: Int) {
     require(k <= numFeatures,
       s"source vector size $numFeatures must be no less than k=$k")
 
+    require(PCAUtil.memoryCost(k, numFeatures) <= Int.MaxValue,
```

Comment:
As long as you're making updates...how about making this strict inequality?  (I could imagine boundary issues with indexing somewhere in Breeze or MLlib.)